src/main/java/bio/overture/ego/service/UserService.java

```diff
@@ -156,22 +157,61 @@ public User createFromIDToken(IDToken idToken) {
             .lastName(idToken.getFamily_name())
             .status(userDefaultsConfig.getDefaultUserStatus())
             .type(userDefaultsConfig.getDefaultUserType())
+            .identityProvider(idToken.getIdentity_provider())
+            .providerId(idToken.getProvider_id())
             .build());
   }
 
   public User getUserByToken(@NonNull IDToken idToken) {
+    val provider = idToken.getIdentity_provider();
+    val providerId = idToken.getProvider_id();
     val userName = idToken.getEmail();
-    val user =
-        findByName(userName)
+
+    User user =
+        findByProviderAndProviderId(provider, providerId)
+            .or(
+                () -> {
+                  if (!isNull(userName)) {
+                    return findByName(userName);
+                  }
+                  log.info("No email provided");
+                  return Optional.empty();
+                })
```

Comment:
following the previous comment, if the flyway migration makes all current user providerTypes `UNKNOWN` and their providerId == email, then in the `.or` statement, we can search by (UNKNOWN, email) and if there is a hit, then we change that record to the providerType and providerId used to sign in.